GPy/mappings/mlp.py

```diff
@@ -47,7 +47,7 @@ def gradients_X(self, dL_dF, X):
 
         # Backpropagation to hidden layer.
         dL_dact = np.dot(dL_dF, self.W2.T)
-        dL_dlayer1 = dL_dact / np.square(np.cosh(layer1))
+        dL_dlayer1 = dL_dact * (1 - np.power(activations, 2))
```

Comment:
Please use np.square instead of np.power(..., 2). In this case, np.square is more efficient.